backend/entityservice/tasks/comparing.py

```diff
@@ -241,91 +292,132 @@ def new_child_span(name, parent_scope=None):
             parent_scope = compute_filter_similarity
         return compute_filter_similarity.tracer.start_active_span(name, child_of=parent_scope.span)
 
-    log.debug("Computing similarity for a chunk of filters")
+    log.debug(f"Computing similarities for {len(package)} chunks of filters")
     log.debug("Checking that the resource exists (in case of run being canceled/deleted)")
     assert_valid_run(project_id, run_id, log)
 
-    chunk_info_dp1, chunk_info_dp2 = chunk_info
-
-    with DBConn() as conn:
-        with new_child_span('fetching-encodings') as parent_scope:
-            with new_child_span('fetching-left-encodings', parent_scope):
-                log.debug("Fetching and deserializing chunk of filters for dataprovider 1")
+    #chunk_info_dp1, chunk_info_dp2 = chunk_info
+    def reindex_using_encoding_ids(recordarray, encoding_id_list):
+        # Map results from "index in chunk" to encoding id.
+        return array.array('I', [encoding_id_list[i] for i in recordarray])
+
+    num_results = 0
+    num_comparisons = 0
+    sim_results = []
+
+    if len(package) > 1:  # multiple full blocks in one package
```

Comment:
I'm surprised to see this isn't the only path - why not just iterate over the packages knowing that sometimes there is only one package (when the whole chunk is part of the same block)?